### PR TITLE
Use context for auth access controllers

### LINKED ISSUE #137: Make User information available via auth package
Currently, authentication doesn't make any user information available to the request context. This is useful for logging, event generation and auditing. It is also necessary to full event notification support.

This work should consist of the following:

1. Define interface involving `context.Context` in authorization interface.
2. Provide a common `User` type to be used across authentication implementations.
3. Provide functions that make user available in context.

### PR DESCRIPTION
**Pull request by jlhawn**
The auth package has been updated to use "golang.org/x/net/context" for
passing information between the application and the auth backend.

AccessControllers should now set a "auth.user" context value to a AuthUser
struct containing a single "Name" field for now with possible, optional, values
in the future.

The "silly" auth backend always sets the name to "silly", while the "token" auth
backend will set the name to match the "subject" claim of the JWT.

closes #137

### CODE REVIEW

File: auth/auth.go [hunk 1]
```diff
@@ -65,13 +88,16 @@ type Challenge interface {
...
 // complete denial and http authorization challenges.
 type AccessController interface {
-	// Authorized returns non-nil if the request is granted access. If one or
-	// more Access structs are provided, the requested access will be compared
-	// with what is available to the request. If the error is non-nil, access
-	// should always be denied. The error may be of type Challenge, in which
-	// case the caller may have the Challenge handle the request or choose
-	// what action to take based on the Challenge header or response status.
-	Authorized(req *http.Request, access ...Access) error
+	// Authorized returns a non-nil error if the request is granted access and
```
Comment (stevvooe) [inline auth/auth.go]: An `AccessController` no longer authorizes a request; it produces an authorized context.

Should read as follows:

```
// Authorized returns a non-nil error if the context is granted access and
// ...
```

**Comment by stevvooe** [comment]
LGTM!

We may want to place the token into the context, along with the access that is available but this is fine for now.

**Comment by dmp42** [comment]
It's failing to find the package.

**Comment by jlhawn** [comment]
@dmp42 that's because we're waiting to merge @stevvooe's work on #86 first

**Comment by stevvooe** [comment]
@jlhawn After thinking about it, there is no reason to hold up the merge of this PR waiting on #86.

Go ahead and add golang.org/x/net/context to godep. Add a `WithRequest` function (I'll move this to a new context package I'm working on):

```
func WithRequest(ctx context.Context, r *http.Request) context.Context {
    return context.WithValue("http.request", r)
}
```

Update app.go to push the request into the context before calling authorized (no reason to propagate context all throughout application and request context, yet):

```
ctx := WithRequest(context.TODO(), r)
if _, err := app.accessController.Authorized(ctx, accessRecords...); err != nil {
...
}
// We don't really need the auth'd context yet
```

This will be compatible with what I'm doing now and should be equivalent to the current behavior.

**Comment by stevvooe** [comment]
LGTM!

**Pull request merged**